core/pkg/eval/json_evaluator.go

```diff
@@ -336,6 +349,37 @@ func (je *JSONEvaluator) evaluateVariant(reqID string, flagKey string, context m
 	return flag.DefaultVariant, flag.Variants, reason, metadata, nil
 }
 
+func (je *JSONEvaluator) setFlagdProperties(
+	context map[string]any,
+	properties flagdProperties,
+) map[string]any {
+	if context == nil {
+		context = map[string]any{}
+	}
+
+	if _, ok := context[flagdPropertiesKey]; ok {
+		je.Logger.Warn("overwriting $flagd properties in the context")
+	}
+
+	context[flagdPropertiesKey] = properties
+
+	return context
+}
+
+func getFlagdProperties(context map[string]any) (flagdProperties, bool) {
+	properties, ok := context[flagdPropertiesKey]
+	if !ok {
+		return flagdProperties{}, false
+	}
+
+	var p flagdProperties
+	if err := mapstructure.Decode(properties, &p); err != nil {
```

Comment:
here we could also use a `json.Marshal/json.Unmarshal`, if we do not want to introduce a new dependency to github.com/mitchellh/mapstructure

Reply:
Oh yeah, that is much nicer. Thank you! Updated.